jetty-server/src/main/java/org/eclipse/jetty/server/OptionalProxySslConnectionFactory.java

```diff
@@ -0,0 +1,112 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.jetty.io.EndPoint;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * <p>A ConnectionFactory whose connections detect whether the first bytes are
+ * proxy or TLS bytes and upgrades to either a TLS connection or to another configurable
+ * connection.</p>
+ */
+public class OptionalProxySslConnectionFactory extends DetectorConnectionFactory
```

Comment:
Why have this Proxy-or-TLS factory?
Isn't the idea of composable factories to have a ProxyFactory and a TlsFactory and one or the other will recognise the bytes.

Reply:
You're right, configuring the connection factories as proxy, optional ssl, http has the exact same effect. Let me get rid of this class.